sympy/solvers/solvers.py

```diff
@@ -1093,6 +1092,13 @@ def _has_piecewise(e):
     ###########################################################################
     if bare_f:
         solution = _solve(f[0], *symbols, **flags)
+        if solution is not None:
+            for solnArgs in solution:
+                if not isinstance(solnArgs,(dict,tuple,list,set)):
+                    if solnArgs.has(LambertW) and solnArgs.has(Lambda):
+                        return solution
+                        break
```

Comment:
It shouldn't be necessary to add this kind of special case code in the main `solve` function.

Reply:
I think it could be removed.